src/PhpSessionPersistence.php

```diff
@@ -105,4 +147,82 @@ private function generateSessionId() : string
     {
         return bin2hex(random_bytes(16));
     }
+
+    /**
+     * Generate cache headers for a given session cache_limiter value.
+     * @param string $cacheLimiter
+     * @param int $cacheExpire
+     */
+    private function generateCacheHeaders(string $cacheLimiter, int $cacheExpire = 0) : array
+    {
+        // Unsupported cache_limiter
+        if (! isset(self::$supported_cache_limiters[$cacheLimiter])) {
+            return [];
+        }
+
+        // cache_limiter: 'nocache'
+        if ('nocache' === $cacheLimiter) {
+            return [
+                'Expires'       => self::CACHE_PAST_DATE,
+                'Cache-Control' => 'no-store, no-cache, must-revalidate',
+                'Pragma'        => 'no-cache',
+            ];
+        }
+
+        $maxAge       = 60 * $cacheExpire;
+        $lastModified = $this->getLastModified($_SERVER['SCRIPT_FILENAME'] ?? '');
```

Comment:
We typically try not to access superglobals directly. Please pass this value to the method by using the following from within `initializeSessionFromRequest()`:

```php
$request->getServerParams()['SCRIPT_FILENAME']
```

Considering that this may not be set, you'll need to come up with a default value to use; in that case, I'd likely use `__FILE__`:

```php
$request->getServerParams()['SCRIPT_FILENAME'] ?? __FILE__
```

Reply:
Yeah, I thought about that too...I was actually waiting for your review. I actually used __FILE__ in the test case

I'd have a few questions before applying the changes:

1. generateCacheHeaders(...) is privately used only with internal properties, should I remove the arguments and use them directly?

2. same question getLastModified, since the scriptFile will be saved into an internal property

3. Should I remove thi PR and re-add a 1-commit PR? (I started uploading to upstream while forgetting i had already opened the PR, I apologize for the number of commits)

Reply:
1. Sure, that should work fine.
2. Same.
3. No, I'm fine with multiple commits. If YOU are not, then run `git rebase -i` and squash some commits and force push back to your branch.